### PR TITLE
486623: Add admin decommission service page

### PR DESCRIPTION
**Pull request by philsegal**
Screenshot 2024-12-13 at 09 18 37
Screenshot 2024-12-13 at 09 18 53

### CODE REVIEW

File: src/server/decommission-service/controllers/decommission-form.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { buildOptions } from '~/src/server/common/helpers/options/build-options.js'
```
Comment (feedmypixel) [inline src/server/decommission-service/controllers/decommission-form.js:1]: So the folders reflect the urls on the frontend. Looks like this is an admin section. So please put this folder in the `admin` folder and update the urls to be in the admin section

File: src/server/decommission-service/controllers/decommission-form.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const decommissionFormController = {
+  handler: async (request, h) => {
+    const query = request?.query
+    const redirectLocation = query?.redirectLocation
+
+    const { repositories } = await fetchRepositories()
+
+    const repositoriesValues = repositories.map((repo) => {
+      return { value: repo.id, text: repo.description }
```
Comment (feedmypixel) [inline src/server/decommission-service/controllers/decommission-form.js:12]: Theres a `buildOptions()` helper that will build these options for you

File: src/server/decommission-service/routes.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { authScope } from '~/src/server/common/helpers/auth/auth-scope.js'
+import { scopes } from '~/src/server/common/constants/scopes.js'
+import { decommissionFormController } from '~/src/server/decommission-service/controllers/decommission-form.js'
+import { decommissionServiceController } from '~/src/server/decommission-service/controllers/decommission-service.js'
+import { summaryController } from '~/src/server/decommission-service/controllers/summary.js'
+import { provideSubNavigation } from '~/src/server/admin/helpers/provide-sub-navigation.js'
+
+const adminScope = authScope([`+${scopes.admin}`])
```
Comment (feedmypixel) [inline src/server/decommission-service/routes.js:8]: with this folder going into the admin section. The entire section is wrapped in admin scope. So you will get this for free

File: src/server/decommission-service/routes.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+const decommissionService = {
+  plugin: {
+    name: 'decommission service',
+    register: (server) => {
+      server.ext([
```
Comment (feedmypixel) [inline src/server/decommission-service/routes.js:14]: this is also handled in the admin section

File: src/server/decommission-service/controllers/decommission-form.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    })
+
+    const repositoriesOptions = buildOptions(repositoriesValues ?? [])
+
+    return h.view('decommission-service/views/decommission-form', {
+      pageTitle: 'Decommission Service',
+      heading: 'Decommission a service',
+      headingCaption: 'Provide the service name to decommission',
+      redirectLocation,
```
Comment (feedmypixel) [inline src/server/decommission-service/controllers/decommission-form.js:21]: not sure why this is being passed around? I suspect this is leftovers from the multi step form work. I don't think you need to pass the redirect location along in the forms here? it doesn't seem to doing anything

File: src/server/decommission-service/helpers/schema/service-validation.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import Joi from 'joi'
+
+import { validation } from '~/src/server/common/constants/validation.js'
+
+function serviceValidation(serviceNames) {
+  // Without clientside js, fallback search
```
Comment (feedmypixel) [inline src/server/decommission-service/helpers/schema/service-validation.js:6]: is this comment needed?

File: src/server/decommission-service/helpers/schema/service-validation.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  return Joi.object({
+    serviceName: Joi.string()
+      .valid(...serviceNames)
+      .required()
+      .messages({
+        'any.only': validation.chooseAnEntry,
+        'any.required': validation.chooseAnEntry,
+        'string.empty': validation.chooseAnEntry
+      }),
+    redirectLocation: Joi.string().valid('summary', '')
```
Comment (feedmypixel) [inline src/server/decommission-service/helpers/schema/service-validation.js:16]: not needed?

**Review by flurdy — APPROVED** [review]

**Pull request merged**